### PR TITLE
[IS 5.2.0] Saml metadata profile for IS

### PR DESCRIPTION
**Pull request by madurangasiriwardena**

### CODE REVIEW

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java [hunk 1]
```diff
@@ -47,13 +47,124 @@ public SAMLSSOConfigAdmin(Registry userRegistry) {
...
+        serviceProviderDTO.setDoSignResponse(serviceProviderDO.isDoSignResponse());
+        serviceProviderDTO.setDoSignAssertions(serviceProviderDO.isDoSignAssertions());
+        serviceProviderDTO.setNameIdClaimUri(serviceProviderDO.getNameIdClaimUri());
+        serviceProviderDTO.setEnableAttributesByDefault(serviceProviderDO.isEnableAttributesByDefault());
+
+        if (serviceProviderDO.getNameIDFormat() == null) {
+            serviceProviderDO.setNameIDFormat(NameIdentifier.EMAIL);
+        } else {
+            serviceProviderDO.setNameIDFormat(serviceProviderDO.getNameIDFormat().replace("/",
+                    ":"));
```
Comment (omindu) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: Can be moved to the line above.

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/upload_service_provider_from_url_finish.jsp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        }else{
+            attributeConsumingServiceIndex = "";
+        }
+
+
+        samlSsoServuceProviderConfigBean.clearBean();
+
+        String message = resourceBundle.getString("sp.added.successfully");
+
+		CarbonUIMessage.sendCarbonUIMessage(message, CarbonUIMessage.INFO, request);
```
Comment (omindu) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/upload_service_provider_from_url_finish.jsp]: Check formatting

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/upload_service_provider_from_url_finish.jsp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+<jsp:setProperty name="samlSsoServuceProviderConfigBean" property="*"/>
+
+<script type="text/javascript" src="global-params.js"></script>
+<script type="text/javascript" src="../carbon/admin/js/breadcrumbs.js"></script>
+<script type="text/javascript" src="../carbon/admin/js/cookies.js"></script>
+<script type="text/javascript" src="../carbon/admin/js/main.js"></script>
+
+
+<%
+	String backendServerURL;
```
Comment (omindu) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/upload_service_provider_from_url_finish.jsp]: Indentation issue

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider.jsp [hunk 4]
```diff
@@ -147,6 +147,28 @@ function doValidation() {
...
+                if (value.length != 0) {
+                    if (!regexp.test(value)) {
+                        CARBON.showWarningDialog(
+                                "<fmt:message key='sp.enter.valid.url'/>",
+                                null, null);
+                        return false;
+                    }
+                }
+            }else{
+                CARBON.showWarningDialog("<fmt:message key='sp.enter.valid.url'/>",null, null);
```
Comment (omindu) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider.jsp]: Formatting issue

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/java/org/wso2/carbon/identity/sso/saml/ui/client/SamlSPMetadataUploadExecutor.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                httpServletResponse.setContentType("text/html; charset=utf-8");
+                msg = "Metadata have been uploaded successfully.";
+                String attributeConsumingServiceIndex = "";
+                if (serviceProviderDTO.getAttributeConsumingServiceIndex() != null){
+                    attributeConsumingServiceIndex = serviceProviderDTO.getAttributeConsumingServiceIndex();
+                }
+
+                CarbonUIMessage.sendCarbonUIMessage(msg, CarbonUIMessage.INFO, httpServletRequest,
+                        httpServletResponse, getContextRoot(httpServletRequest)
+                                + "/" + webContext + "/application/configure-service-provider.jsp?action=update&display=samlIssuer&spName="+spName+"&samlIssuer="+serviceProviderDTO.getIssuer()+"&attrConServIndex="+attributeConsumingServiceIndex);
```
Comment (omindu) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/java/org/wso2/carbon/identity/sso/saml/ui/client/SamlSPMetadataUploadExecutor.java]: Formatting issue.

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/java/org/wso2/carbon/identity/sso/saml/ui/client/SAMLSSOConfigServiceClient.java [hunk 6]
```diff
@@ -52,13 +60,54 @@ public SAMLSSOConfigServiceClient(String cookie, String backendServerURL, Config
...
+        return serviceProviderDTO;
+    }
+
+    public boolean updateServiceProvider(SAMLSSOServiceProviderDTO serviceProviderDTO) throws
+            IdentitySAML2SSOUiException {
+
+        boolean status;
+        try {
+            status = stub.updateRPServiceProvider(serviceProviderDTO);
+        } catch (RemoteException  | IdentitySAMLSSOConfigServiceIdentitySAML2SSOException e) {
```
Comment (omindu) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/java/org/wso2/carbon/identity/sso/saml/ui/client/SAMLSSOConfigServiceClient.java]: Formatting issue - extra space

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java [hunk 7]
```diff
@@ -104,75 +350,232 @@ protected SAMLSSOServiceProviderDO resourceToObject(Resource resource) {
...
+            assertionConsumerService.setBinding(SAMLConstants.SAML2_POST_BINDING_URI);
+            assertionConsumerService.setLocation(samlssoServiceProviderDO.getAssertionConsumerUrl());
+            spSSODescriptor.getAssertionConsumerServices().add(assertionConsumerService);
+            entityDescriptor.getRoleDescriptors().add(spSSODescriptor);
+
+            //Single Logout
+            if (samlssoServiceProviderDO.isDoSingleLogout()) {
+                SingleLogoutService singleLogoutService = createSAMLObject(SingleLogoutService.class);
+                singleLogoutService.setBinding(SAMLConstants.SAML2_POST_BINDING_URI);
+                if (samlssoServiceProviderDO.getLogoutURL() != null && !samlssoServiceProviderDO.getLogoutURL()
```
Comment (omindu) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java]: Use StringUtils.isNotBlank

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java [hunk 8]
```diff
@@ -104,75 +350,232 @@ protected SAMLSSOServiceProviderDO resourceToObject(Resource resource) {
...
+                        .equals("")) {
+                    singleLogoutService.setLocation(samlssoServiceProviderDO.getLogoutURL());
+                } else {
+                    singleLogoutService.setLocation(samlssoServiceProviderDO.getAssertionConsumerUrl());
+                }
+                spSSODescriptor.getSingleLogoutServices().add(singleLogoutService);
+            }
+
+            //Certificate Alias
+            if (samlssoServiceProviderDO.getCertAlias() != null
```
Comment (omindu) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java]: Use StringUtils.isNotBlank

File: components/identity-core/org.wso2.carbon.identity.core/pom.xml [hunk 9]
```diff
@@ -78,6 +82,11 @@
             <groupId>org.wso2.securevault</groupId>
             <artifactId>org.wso2.securevault</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.opensaml</groupId>
+            <artifactId>opensaml</artifactId>
+            <version>2.6.4</version>
```
Comment (Kishanthan) [inline components/identity-core/org.wso2.carbon.identity.core/pom.xml]: Don't hardcode the version. All dependency versions should be from a parent dependencyManagement section.

File: components/identity-core/org.wso2.carbon.identity.core/pom.xml [hunk 10]
```diff
@@ -132,10 +142,20 @@
...
                             org.wso2.carbon.identity.base;version="${carbon.identity.package.import.version.range}",
                             org.wso2.carbon.security.config;version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.security; version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.security.keystore; version="${carbon.identity.package.import.version.range}",
+                            org.wso2.carbon.security.keystore.service; version="${carbon.identity.package.import.version.range}",
 
                             org.wso2.carbon.registry.core.*;version="${carbon.kernel.registry.imp.pkg.version}",
 
                             org.wso2.carbon.user.api; version="${carbon.user.api.imp.pkg.version.range}",
+                            javax.xml.parsers,
```
Comment (Kishanthan) [inline components/identity-core/org.wso2.carbon.identity.core/pom.xml]: Import packages should have import version range.

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java [hunk 11]
```diff
@@ -39,33 +122,210 @@ public SAMLSSOServiceProviderDAO(Registry registry) {
...
 
-        if (resource
-                .getProperty(IdentityRegistryResources.PROP_SAML_SSO_USE_FULLY_QUALIFIED_USERNAME_AS_SUBJECT) != null) {
-            serviceProviderDO
-                    .setUseFullyQualifiedUsername(new Boolean(
-                            resource.getProperty(
-                                    IdentityRegistryResources.PROP_SAML_SSO_USE_FULLY_QUALIFIED_USERNAME_AS_SUBJECT)
-                                    .trim()));
+        try {
+            Configuration.registerObjectProvider(UseFullyQualifiedUsername.DEFAULT_ELEMENT_NAME, new
```
Comment (Kishanthan) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOServiceProviderDAO.java]: Normally, as a best practice and for readability, we use line break symbols or characters such as comma or full stop punctuation (',' , '.') to break a line go to the next within our java source. Please fix these accordingly throughout all your PRs.

For example, the above line , you can break the line into multiple like the following. Configuration.registerObjectProvider(UseFullyQualifiedUsername.DEFAULT_ELEMENT_NAME,
                new UseFullyQualifiedUsernameBuilder(), new UseFullyQualifiedUsernameMarshaller(),
                new UseFullyQualifiedUsernameUnmarshaller());

**Pull request closed without merging**